Dashboard/app/js/lib/components/devices/AssignmentsList.jsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+/* eslint-disable no-nested-ternary */
+const AssignmentsList = ({
+  assignments,
+  onEdit,
+  onDelete,
+  onSort,
+}) => (
+  <div>
+    <div className="deviceControls">
+      <a
+        className="btnOutline"
+        onClick={() => onEdit('new')}
+        onKeyDown={() => onEdit('new')}
+        style={{ float: 'right' }}
+      >
+        {Ember.String.loc('_create_new_assignment')}
+      </a>
+    </div>
+    <div
+      id="devicesListTable_length"
+      className="dataTables_length"
+    />
+    {!assignments.get('length') && (
+      <div style={{ marginTop: '40px' }}>No assignments</div>
+    )}
+    {assignments.get('length') > 0 && (
+      <table className="dataTable">
+        {/* TABLE HEADER */}
+        <thead>
+          <tr>
+            <th
+              className={FLOW.tableColumnControl.get('selected') == 'name' ? (FLOW.tableColumnControl.get('sortAscending') ? 'sorting_asc' : 'sorting_desc') : ''}
+            >
+              <a
+                onClick={() => onSort('name')}
+                onKeyDown={() => onSort('name')}
+              >
+                {Ember.String.loc('_name')}
```

Comment:
NB: try to remove all references to Ember from inside these react components because that way we can decouple the components from ember and they can be reused elsewhere - for example in Lumen which is not an Ember app.
Pass the end value to be rendered it as a prop instead